src/js/os.js

```diff
@@ -88,3 +88,18 @@ Object.defineProperty(exports, 'EOL', {
     return '\n';
   }
 });
+
+function isStringEndsWith(str, search) {
+  var len = str.length;
+  return str.substring(len - search.length, len) === search;
+}
+
+exports.tmpdir = function tmpdir() {
+  var path = process.env.TMPDIR ||
```

Comment:
why not get env from [getenv(3)](http://man7.org/linux/man-pages/man3/getenv.3.html)

Reply:
What's the benefits in that way? Actually we did set the env at [`GetEnvironArray`](https://github.com/yodaos-project/ShadowNode/blob/master/src/modules/iotjs_module_process.c#L423) via [`environ(7)`](http://man7.org/linux/man-pages/man7/environ.7.html).

Reply:
in Windows, Node.js is also in this way，but on other [platforms](https://github.com/nodejs/node/blob/908292cf1f551c614a733d858528ffb13fb3a524/lib/os.js#L133) are via [getenv](https://github.com/nodejs/node/blob/a8277ddb5505056e3d66d02051c9195310c5c715/src/node_credentials.cc#L61), i am not sure why Node.js is doing this.

Reply:
getenv is only used if node environment has not been setup.

Reply:
See https://github.com/nodejs/node/pull/18511 for all related history, this seems a historic design problem that allows packages sharing via `process.env`(write/read), but anyway, we have to handle this security, thanks @lolBig